Only process dispatches for your port

collinsauve opened the pull request:
See #71

vhmth commented on test/Store.test.js:
```diff
@@ -214,6 +214,7 @@ describe('Store', function() {
       spy.calledOnce.should.eql(true);
       spy.alwaysCalledWith('xxxxxxxxxxxx', {
         type: DISPATCH_TYPE,
+        port: 'test',
```
Can you make a this a constant at the top of the test file?

collinsauve replied:
Can do

vhmth commented on test/Store.test.js:
```diff
@@ -227,6 +228,7 @@ describe('Store', function() {
       spy.calledOnce.should.eql(true);
       spy.alwaysCalledWith('', {
```
Could you add a test suite that explictly sets up two stores, dispatches an action to each, and ensures that we only dispatch to proper port names?

collinsauve replied:
I'll give it a shot, but honestly I'm not really sure where to start.   Maybe if I stare at this test code long enough it will come to me :)

vhmth replied:
https://www.useloom.com/share/8d2816a508f34141b76fe0e1a0ac708b

collinsauve replied:
Hi Vinay,

I watched your video message, and I took a deeper look into the tests as they are.  This test already spys `sendMessage` and ensures that the message is sent with the `portName`:

> it('should send a message with the correct dispatch type and payload given an extensionId',)

I think that I need to test that `wrapStore` behaves correctly instead?   The missing change that is not tested is that listener `dispatchResponse` created by `wrapStore` should discard messages that do not include its port.

However I do not see any `wrapStore` tests at all!   Shall I try to create some?

collinsauve replied:
I've added another commit (e5b9ce7) with 2 tests around `wrapStore`

vhmth replied:
🎉 heck yes! Good call @collinsauve

vhmth commented:
Great catch 🎉

vhmth commented on test/wrapStore.test.js:
```diff
@@ -0,0 +1,110 @@
+
+describe('wrapStore', function() {
+    const portName = 'test';
+
```
Could we get rid of this whitespace?

vhmth replied:
Just one new line from `const`

collinsauve replied:
done.

vhmth commented:
@collinsauve I think we should wait for https://github.com/tshaddix/react-chrome-redux/pull/73 to land so you can rebase off it and lint - good stuff on these tests!

vhmth commented:
Cool! Just waiting for the lint PR to get landed before following up here for the rebase. :-)

vhmth commented:
Hey @collinsauve could you try rebasing and linting the tests you added? Appreciate the patience here. :-)

collinsauve commented:
Just working through merge conflicts now, unfortunately I can't run the tests locally (https://github.com/tshaddix/react-chrome-redux/issues/74) so I apologize for abusing your travis to do it for mee

vhmth commented:
Haha abuse him all you want! Looking at https://github.com/tshaddix/react-chrome-redux/issues/74 right now

tshaddix commented on src/store/Store.js:
```diff
@@ -95,9 +96,9 @@ class Store {
         this.extensionId,
         {
           type: DISPATCH_TYPE,
+          port: this.portName,
```
I think we should call this key `portName` rather than `port`. It's more descriptive on what this actually is (especially because `port` is a reference to a full object in other parts of the app)

tshaddix commented on src/wrap-store/wrapStore.js:
```diff
@@ -44,7 +44,7 @@ export default (store, {
    */
   const dispatchResponse = (request, sender, sendResponse) => {
-    if (request.type === DISPATCH_TYPE) {
+    if (request.type === DISPATCH_TYPE && request.port === portName) {
```
`request.portName` w/ name change

tshaddix commented on test/Store.test.js:
```diff
@@ -207,26 +209,28 @@ describe('Store', function () {
       spy.calledOnce.should.eql(true);
       spy.alwaysCalledWith('xxxxxxxxxxxx', {
         type: DISPATCH_TYPE,
+        port: portName,
```
`portName`

tshaddix commented on test/Store.test.js:
```diff
@@ -207,26 +209,28 @@ describe('Store', function () {
       spy.calledOnce.should.eql(true);
       spy.alwaysCalledWith('', {
         type: DISPATCH_TYPE,
+        port: portName,
```
`portName`

tshaddix commented on test/wrapStore.test.js:
```diff
@@ -0,0 +1,114 @@
+    };
+    const message = {
+      type: DISPATCH_TYPE,
+      port: portName,
```
`portName`

tshaddix commented on test/wrapStore.test.js:
```diff
@@ -0,0 +1,114 @@
+      };
+      const message = {
+        type: DISPATCH_TYPE,
+        port: portName + '2',
```
`portName`

vhmth commented:
Hey @collinsauve just making sure but have you rebased and linted?

collinsauve commented:
@vhmth I had rebased and used your Travis to lint for me on March 24th.  It is difficult for me to lint locally due to #74.

I've rebased again and repushed just now, and Travis passed.

tshaddix commented:
@collinsauve nothing for you to do on my end. Just going to do another review real quick. Thanks for your patience!

tshaddix commented:
@vhmth We good to merge?

vhmth commented:
Woah completely missed that. Sorry @collinsauve! Yes this is good to merge @tshaddix :-)

tshaddix commented:
Awesome @vhmth, merging now. Thanks @collinsauve !
